### PR TITLE
added api test for hostgroup clone, related to #2476

### PR DESCRIPTION
**Pull request by pondrejk**
test result:
```
nosetests tests/foreman/api/test_hostgroup.py:HostGroupTestCase.test_positive_clone -v
Create a hostgroup by cloning an existing one ... ok

----------------------------------------------------------------------
Ran 1 test in 14.172s

OK
 ```

### CODE REVIEW

File: tests/foreman/api/test_hostgroup.py [hunk 1]
```diff
@@ -198,6 +198,24 @@ def test_positive_create_with_name(self):
...
+
+        @id: 44ac8b3b-9cb0-4a9e-ad9b-2c67b2411922
+
+        @assert: A hostgroup is cloned with new name
+        """
+        hostgroup = entities.HostGroup(
+            location=[self.loc],
+            organization=[self.org],
+        ).create()
+        hostgroup_cloned_name = gen_string('alpha', 6)
```
Comment (svtkachenko) [inline tests/foreman/api/test_hostgroup.py]: There is no need to limit new name length here, isn't it?

File: tests/foreman/api/test_hostgroup.py [hunk 2]
```diff
@@ -198,6 +198,24 @@ def test_positive_create_with_name(self):
...
+        hostgroup_cloned = entities.HostGroup(
+                id=hostgroup.id
+                ).clone(data={u'name': hostgroup_cloned_name})
+        self.assertEqual(hostgroup_cloned[u'name'], hostgroup_cloned_name)
```
Comment (svtkachenko) [inline tests/foreman/api/test_hostgroup.py]: I think it is not required to cast unicode on dict key name.

Comment (sghai) [inline tests/foreman/api/test_hostgroup.py]: I think its good to have a `clone` helper function in nailgun itself ?

Comment (oshtaier) [inline tests/foreman/api/test_hostgroup.py]: Yeah, `[u'name']` seems a little bit weird

File: tests/foreman/api/test_hostgroup.py [hunk 3]
```diff
@@ -198,6 +198,24 @@ def test_positive_create_with_name(self):
...
+        @id: 44ac8b3b-9cb0-4a9e-ad9b-2c67b2411922
+
+        @assert: A hostgroup is cloned with new name
+        """
+        hostgroup = entities.HostGroup(
+            location=[self.loc],
+            organization=[self.org],
+        ).create()
+        hostgroup_cloned_name = gen_string('alpha', 6)
+        hostgroup_cloned = entities.HostGroup(
```
Comment (oshtaier) [inline tests/foreman/api/test_hostgroup.py]:
```
hostgroup_cloned = entities.HostGroup(
    id=hostgroup.id
).clone(data={u'name': hostgroup_cloned_name})
```
seems as better format for that block

**Comment by pondrejk** [comment]
All comments addressed.

**Comment by sghai** [comment]
@oshtaier @svtkachenko : guys, can we merge this if all your comments have been addressed ?

**Comment by oshtaier** [comment]
Yes

**Pull request merged**